premarum-backend/PreEnrollmentMgmt.Core/Services/StudentService.cs

```diff
@@ -0,0 +1,30 @@
+using PreEnrollmentMgmt.Core.Entities;
+using PreEnrollmentMgmt.Core.Exceptions;
+using PreEnrollmentMgmt.Core.Repositories;
+
+namespace PreEnrollmentMgmt.Core.Services;
+
+public class StudentService
+{
+    
+    private readonly IStudentRepository _studentRepository;
+    private readonly StudentValidationService _studentValidationService;
+
+    public StudentService( IStudentRepository studentRepository,
+        StudentValidationService studentValidationService)
+    {
+        _studentRepository = studentRepository;
+        _studentValidationService = studentValidationService;
+    }
+
+    public async Task<Student> GetStudent(string studentEmail)
+    {
+        return await _studentValidationService.ValidateStudentExists(studentEmail);
```

Comment:
You need to create student if it does not exist

Reply:
Discuss in next meeting some questions about the Id for a student